File: tuan02/bai36/bai36.jl
    function giai_thua(n::Int64)
        if n < 0
            return Int64(0), false
        end
        kq = Int64(1)
        llong_max = typemax(Int64)
        for i in 1:n
            if div(llong_max, i) < kq
                return Int64(0), false # Bị tràn số
            end
            kq *= i
        end
        return kq, true
    end

    input_str = readline()
    if !isempty(input_str)
        n = parse(Int64, input_str)
        kq, ok = giai_thua(n)
        if ok
            println(kq)
        else
            println("Tran so")
        end
    end

File: tuan02/bai36/bai36.cpp
#include <iostream>
#include <climits>

using namespace std;

bool giai_thua(int n, long long &kq) {
    if (n < 0) return false;
    kq = 1;
    for (int i = 1; i <= n; i++) {
        if (LLONG_MAX / i < kq) {
            return false; // Bị tràn số
        }
        kq *= i;
    }
    return true;
}

int main() {
    int n;
    if (cin >> n) {
        long long kq;
        if (giai_thua(n, kq)) {
            cout << kq << endl;
        } else {
            cout << "Tran so" << endl;
        }
    }
    return 0;
}